test/e2e-helm-new/e2e_suite.go

```diff
@@ -0,0 +1,159 @@
+			err := tc.Init(
+				"--plugins", "helm.operator-sdk.io/v1",
+				"--project-version", "3-alpha",
+				"--domain", tc.Domain)
+			Expect(err).Should(Succeed())
+
+			By("creating an API definition")
+			err = tc.CreateAPI(
+				"--group", tc.Group,
+				"--version", tc.Version,
+				"--kind", tc.Kind)
+			Expect(err).Should(Succeed())
+
+			By("replace Dockerfile to use the dev tag")
+			version := strings.TrimSuffix(version.Version, "+git")
+			testutils.ReplaceInFile(filepath.Join(tc.Dir, "Dockerfile"), version, "dev")
+
+			By("building the operator image")
+			err = tc.Make("docker-build", "IMG="+tc.ImageName)
+			Expect(err).Should(Succeed())
+
+			By("loading the operator image into the test cluster")
+			err = tc.LoadImageToKindCluster()
+			Expect(err).Should(Succeed())
+
+			By("install CRDs")
+			err = tc.Make("install")
+			Expect(err).Should(Succeed())
+
+			By("deploying the controller manager")
+			err = tc.Make("deploy", "IMG="+tc.ImageName)
+			Expect(err).Should(Succeed())
+
+			By("ensuring the controller-manager pod is running as expected")
+			verifyControllerUp := func() error {
+				// Get pod name
+				podOutput, err := tc.Kubectl.Get(
+					true,
+					"pods", "-l", "control-plane=controller-manager",
+					"-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}"+
+						"{{ \"\\n\" }}{{ end }}{{ end }}")
+				Expect(err).NotTo(HaveOccurred())
+				podNames := kbtestutils.GetNonEmptyLines(podOutput)
+				if len(podNames) != 1 {
+					return fmt.Errorf("expect 1 controller pods running, but got %d", len(podNames))
+				}
+				controllerPodName = podNames[0]
+				Expect(controllerPodName).Should(ContainSubstring("controller-manager"))
+
+				// Validate pod status
+				status, err := tc.Kubectl.Get(
+					true,
+					"pods", controllerPodName, "-o", "jsonpath={.status.phase}")
+				Expect(err).NotTo(HaveOccurred())
+				if status != "Running" {
+					return fmt.Errorf("controller pod in %s status", status)
+				}
+				return nil
+			}
+			Eventually(verifyControllerUp, time.Minute, time.Second).Should(Succeed())
+
+			By("creating an instance of CR")
+			// currently controller-runtime doesn't provide a readiness probe, we retry a few times
+			// we can change it to probe the readiness endpoint after CR supports it.
+			sampleFile := filepath.Join("config", "samples",
+				fmt.Sprintf("%s_%s_%s.yaml", tc.Group, tc.Version, strings.ToLower(tc.Kind)))
+			Eventually(func() error {
+				_, err = tc.Kubectl.Apply(true, "-f", sampleFile)
+				return err
+			}, time.Minute, time.Second).Should(Succeed())
+
+			By("ensuring the created resource object gets reconciled in controller")
+			managerContainerLogs := func() string {
+				logOutput, err := tc.Kubectl.Logs(controllerPodName, "-c", "manager")
+				Expect(err).NotTo(HaveOccurred())
+				return logOutput
+			}
+			Eventually(managerContainerLogs, time.Minute, time.Second).Should(ContainSubstring("Reconciled release"))
+
```

Comment:
We should port all the checks in the bash-based e2e test to this test as well (with the exception of the migrate stuff)

Reply:
I did not get this suggestion. Should we do the same test twice? (in go and shell) if yes, why?

Reply:
The shell-based test is testing the legacy layout and the go-based test is testing the KB layout. I think we probably do want to repeat all of the tests to make sure all of the scenarios are tested with the new layout. Before 1.0, we'll be removing the legacy code and legacy e2e tests.

Reply:
I would say the e2e updates should not block this PR from merging, but if you're waiting on reviews, it would be good to work on them in this PR.

Reply:
I already started to work on it, it worked 100% locally adapted to run in the minikube. The problem is for me to make it works in the CI/kind. :-( But it is tracked in the JIRA as subtask and here as follow up.

Reply:
Done but we will work as follow up doing the suggestion made.